### DISCUSSION TITLE
How to display makie window when calling script from the terminal?

### DISCUSSION THREAD

**Discussion by GrayJack** [JuliaPlots/Makie.jl#1861] [Q&A] [answered]
When I'm using VSCode and running line-by-line the window opens, but when I run `julia myscript.jl` on the terminal the window doesn't show up. So how can I make the widows show up when running from the terminal?

**Comment by asinghvi17** [comment, marked as answer]
If the code is the same, you could try `display(fig)`  where `fig` is a `Figure`.  If GLMakie is loaded, this should cause an interactive window to pop up; if WGLMakie is loaded it will open a browser window.

**Reply by GrayJack** [reply]
Thanks for the answer. For me is GLMakie, I tried `display(fig)`, the window showed up but before finishing loading the contents of the window, the window closed and the program finished without any error messages. Did you get any behaviors like that?

Also, does changing to WGLMakie is to just change the library, like the API is the same?

**Reply by asinghvi17** [reply]
> the window closed and the program finished without any error messages

That's because your script exits after the figure is fully displayed.  You could wait on the screen to close using the following code at the end of your program:
```julia
wait(display(fig))
```
`display` returns the screen which the figure is plotted to, and `wait` waits for that screen to close before moving on.


> Also, does changing to WGLMakie is to just change the library, like the API is the same?

Yep, same for CairoMakie if you want to save as PDF or SVG.  Makie handles all high-level abstractions, and in general use you probably won't interact with the backends at all beyond the generic `*Makie.activate!()` functions.  The backend only plots figures to the screen.

**Reply by GrayJack** [reply]
Thanks! That solves this issue!

**Comment by Moelf** [comment]
I wonder for `CairoMakie`, if we should save to a file under `/tmp` and open browser on that file, since browser can render svg and png just fine, instead of going to `ElectronDisplay.jl` as currently suggested, which is just a browser but need a package...

**Reply by asinghvi17** [reply]
AFAIK we already do save to `CairoMakie/display.png` etc depending on the backend settings.  I guess it's just a matter of opening the browser if available and requested.

**Reply by SimonDanisch** [reply]
We could put this https://github.com/SimonDanisch/JSServe.jl/blob/master/src/browser_display.jl#L35= into a lightweight package, or copy it to makie for now...
